SwiftCompilerSources/Sources/SIL/Instruction.swift

```diff
@@ -353,7 +353,36 @@ final public class HopToExecutorInst : Instruction, UnaryInstruction {}
 
 final public class FixLifetimeInst : Instruction, UnaryInstruction {}
 
-final public class DebugValueInst : Instruction, UnaryInstruction {}
+// VarDecl is a struct wrapper around a C++ VarDecl pointer. This insulates the Swift interface from the C++ interface and avoids the need to fake a Swift class. When the AST exposes VarDecl, then this can be replaced whenever it is convenient.
+public struct VarDecl {
+  var bridged: BridgedVarDecl
+  
+  public init?(bridged: OptionalBridgedVarDecl) {
+    guard let decl = bridged.decl else { return nil }
+    self.bridged = BridgedVarDecl(decl: decl)
+  }
+  
+  public var userFacingName: String { String(bridged.getUserFacingName()) }
+}
+
+// See C++ VarDeclCarryingInst
+public protocol VarDeclInst {
+  var varDecl: VarDecl? { get }
+}
+
+public protocol DebugVariableInst : VarDeclInst {
+  var debugVariable: OptionalBridgedSILDebugVariable { get }
```

Comment:
I assume you didn't wrap `OptionalBridgedSILDebugVariable` into a  swift type because it will just passed to other functions. Can you add a typealias, e.g. `typealias DebugVariable = OptionalBridgedSILDebugVariable` and use that in the APIs? I guess, the "Optional" prefix is not relevant on the swift side.